### ISSUE TITLE
Poor performance in home network

### ISSUE THREAD

**Issue by githubX82jKm** [chocolatey/choco#1748]
In my home network (LAN) choco.exe seems to have a poor performance. A "choco upgrade all" command waits for 25 seconds before checking the installed packages.

The same command from the same computer when using my mobile phone as a hotspot (LTE) waits less than a second before checking packages.

Problem affects all my computers in my home network. Internet access ist reliable and fast (90 mbit/s downstream, 30 mbit/s upstream, ping of 15 ms).

I have log files for bose szenarios (with --debug --verbose option).

[choco_upgrade_LAN.log](https://github.com/chocolatey/choco/files/2949354/choco_upgrade_LAN.log)
[choco_upgrade_HOTSPOT.log](https://github.com/chocolatey/choco/files/2949355/choco_upgrade_HOTSPOT.log)

Can someone help to identify the problem?
Thanks, Julius

**Comment by bcurran3** [comment]
Hi Julius,

I did take a look at your logs, unfortunately the only thing there of note is the time things took.

Something in your environment is the problem, this isn't a Chocolatey problem.

There's no clear answer to the problem, you're going to have to do some diagnostics.

Things I'd look into:
* do a tracert chocolatey.org connected from each network
* shut off all other computers and network devices on the home LAN and test (or pull the cables from the router) - some other computer might be congesting your network (streaming HD/4K movie, torrents, controlled by a bot, etc)
* Change your DNS - try 1.1.1.1 (CloudFlare) or 8.8.8.8 (Google) or 208.67.222.222 (OpenDNS)

That's what I'd first do. 

My bet is on another computer on the LAN slowing things down.

Good luck and be sure to report back to us.

**Comment by githubX82jKm** [comment]
Hello,

thank you for the fast answer. The problem seems to occur only if IPv6 is activated in my Fritzbox 7490.
When I deactivate this "feature", everything is running very well.

Regards,
Julius

**Comment by ferventcoder** [comment]
Howdy - I thought we had something noted on IPv6 somewhere. in our documentation. I don't find it on a quick search though, I've added the documentation tag so we have it open until we've documented this in https://chocolatey.org/docs/troubleshooting

**Comment by githubX82jKm** [comment]
Although it may be in the documentation, I don't think that many users who have the same problem just can imagine that this is a IPv6 issue. And IPv6 is activated in the Fritzbox as default.

Anyway, many thanks for your fast help and suggestions.

**Comment by githubX82jKm** [comment]
Sorry, it's me again. I noticed that without IPv6 enabled, some internet pages are not reachable anymore (e. g. https://avm.de). So I have to enable IPv6 in my Fritzbox again which leads to the problem with choco.exe again... :-(

**Comment by bcurran3** [comment]
@githubX82jKm 

Julius,

Thanks for the update. Good job figuring it out.

I'm guessing IPv6 is first in your protocol stack. Setting IPv4 as first might fix your problem. (One of those thing you need to test and let us know!)

Take a look at this:
https://support.microsoft.com/en-us/help/929852/guidance-for-configuring-ipv6-in-windows-for-advanced-users
...and try the **Prefer IPv4 over IPv6 in prefix policies**

Fingers crossed this works for you.

There used to be an easy way to change the priorities in the protocol stack but I'll be damned if I can find it now in Windows 10 1809! So the above looks like the easiest way.

Let us know!

**Comment by githubX82jKm** [comment]
Yes, setting **Prefer IPv4 over IPv6** in the registry from your linked article solves the problem as well. So there is no need to disable IPv6 in the Fritzbox completely.

Forgot to mention that OS was Windows 10 1809 (on all tested computers).

Thank you once again.
Best regards,
Julius

**Comment by bcurran3** [comment]
@githubX82jKm 

Julius,

Excellent news. Glad it worked out...  and thanks for the follow-up.

@ferventcoder 

This resolution to the problem should be mentioned in the docs.

**Comment by ferventcoder** [comment]
@bcurran3 it's popped up before - that's why I thought we had documented in troubleshooting. Thanks for the article link, that will be helpful in documenting. Hopefully we can get something into choco to maybe do some detection and note this. We did just add a global validation concept.

**Comment by pauby** [comment]
Thanks for raising this issue. After reviewing it, it's not something we expect to prioritise in the foreseeable future. We’re closing it to keep our backlog focused, but we’re open to revisiting it in the future if priorities shift or new interest arises. You are welcome to comment here or open a new issue if you'd like to discuss it further down the line.
